data/env/snapd.fish.in

```diff
@@ -8,7 +8,14 @@ end
 # looked for in XDG_DATA_DIRS; make sure it includes the relevant directory for
 # snappy applications' desktop files.
 set -ul snap_xdg_path /var/lib/snapd/desktop
-set --path --export XDG_DATA_DIRS $XDG_DATA_DIRS
-if not contains $snap_xdg_path $XDG_DATA_DIRS
-    set XDG_DATA_DIRS $XDG_DATA_DIRS $snap_xdg_path
+# note, snapd supports distros going back as far as Ubuntu 16.04, what means we
+# must make sure that this is compatible with old fish shells which do not have
+# fish_add_path or set --path
+set --export XDG_DATA_DIRS $XDG_DATA_DIRS
+if not contains $snap_xdg_path (string split : "$XDG_DATA_DIRS")
+    if test (string length "$XDG_DATA_DIRS") -gt 0
```

Comment:
This entire block can just be

```fish
set XDG_DATA_DIRS (string join : -- $XDG_DATA_DIRS $snap_xdg_path)
```

If $XDG_DATA_DIRS is empty, it will be elided and only $snap_xdg_path remains.

Or alternatively replace the if-condition:

```fish
if set -q XDG_DATA_DIRS[1]
```